src/client/components/pages/parts/admin-panel-search-results.tsx

```diff
@@ -0,0 +1,160 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PrivilegeBadges from './privilege-badges';
+import PrivsEditModal from './privs-edit-modal';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {faPencilAlt} from '@fortawesome/free-solid-svg-icons';
+import {genEntityIconHTMLElement} from '../../../helpers/entity';
+import {getPrivilegeShieldIcon} from '../../../../common/helpers/privileges-utils';
+
+
+const {Button, Table} = bootstrap;
+
+type AdminPanelSearchResultsState = {
+	selectedUser?: Record<string, unknown>,
+	showModal: boolean
+};
+type AdminPanelSearchResultsProps = {
+	results?: any[],
+	updateResultsOnPrivsChange: () => void,
+	user: Record<string, unknown>
+};
+
+
+/**
+ * Renders the document and displays the 'AdminPanelSearchResults' page.
+ * @returns {ReactElement} a HTML document which displays the AdminPanelSearchResults.
+ * @param {object} props - Properties passed to the component.
+ */
+class AdminPanelSearchResults extends React.Component<AdminPanelSearchResultsProps, AdminPanelSearchResultsState> {
+	static displayName = 'AdminPanelSearchResults';
+
+	static propTypes = {
+		results: PropTypes.array,
+		updateResultsOnPrivsChange: PropTypes.func.isRequired,
+		user: PropTypes.object.isRequired
+	};
+
+	static defaultProps = {
+		results: null
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			selectedUser: null,
+			showModal: false
+		};
+		this.onCloseModal = this.onCloseModal.bind(this);
+	}
+
+	onCloseModal() {
+		this.setState({showModal: false});
+	}
+
+	openPrivsEditModal(user) {
+		this.setState({
+			selectedUser: user,
+			showModal: true
+		});
+	}
+
+	render() {
+		const noResults = !this.props.results || this.props.results.length === 0;
+		const results = this.props.results.map((result) => {
+			if (!result) {
+				return null;
+			}
+			const name = result.defaultAlias ? result.defaultAlias.name :
+				'(unnamed)';
+			const link = `/editor/${result.bbid}`;
+
+			/* eslint-disable react/jsx-no-bind */
+			return (
+				<tr key={result.bbid}>
+					<td>
+						<a href={link}>
+							{genEntityIconHTMLElement('editor')}
+							{name}
+						</a>
+					</td>
+					<td>
+						<PrivilegeBadges privs={result.privs}/>
+					</td>
+					<td>
+						<Button
+							variant="link"
+							onClick={() => this.openPrivsEditModal(result)}
+						>
+							<img height="18" src={getPrivilegeShieldIcon(result.privs)}/> {' '}
+							<FontAwesomeIcon icon={faPencilAlt}/>
```

Comment:
I wonder about this; it seems to me that the shield icon only repeats the information we already have in the privileges column.

I think we only need an "Edit" button for now (in the future maybe "Block"/"Delete" buttons as well), for which I'd rather have a text button that the very small icon which is harder to click on.

What do you think?
Here it is with text, and some button classes (btn-sm for size and btn-success for style)
![image](https://github.com/metabrainz/bookbrainz-site/assets/6179856/432793eb-af9e-446e-8aa1-bc4e3c5e871a)

Reply:
The shield icons were never really for any kind of additional information. They were just an aesthetic choice, one which I dont mind rescinding. One other reason I clubbed them with the pencil icon was so I can increase the hitbox of the click.

I did give some thoughts on how the edit button should look like and here's why I ditched this simple Edit button. The main reason is because I think these buttons, when iterated in a table look really out of place to me, mostly because these types of buttons have mostly been used throughout the site as singular action initiators(i made that up), basically you find them mostly in isolation (as a 'Submit' button in a page, or a 'Create a collection' button, etc). I find that they look a bit odd when iterated in a table:
![Screenshot 2023-06-27 at 16-44-39 Search Results – BookBrainz](https://github.com/metabrainz/bookbrainz-site/assets/56965261/ead9dec3-ee50-4450-8a9b-039b45e3ee73)

This also reminds me that I've been in a similar dilemma before, in this old [PR](https://github.com/metabrainz/bookbrainz-site/pull/830) where I used a question mark as a clickable link. I do agree that it does not work very good as a call to action, but now you know why I went for it instead of a button.

I do realise that this can all be a very subjective discussion, and it is quite possible that I may be reading too much into this. I am okay with changing it to an 'Edit' button as suggested for now, but I think we should go through some more ideas on what kind of buttons we'll use in a table.

Reply:
> The shield icons were never really for any kind of additional information.
I think they look neat, but in this case they should be used in another place if they repeat info from the previous column. Maybe they would be a good indicator in front of user names on the website? Possibly only showing admins and privileged users, and not showing the empty shield icon in front of every user's name.

For this table, if you were to add it, I would add it in front of the user name instead of the user icon (which currently doesn't add any info). Might be helpful when quickly reading through the table.


I get your thought process; the repeated buttons are perhaps not ideal, but IMO they are more *usable* that just the icon (as you mention the hitbox is small) even if they look better. The issue will compound once we have more actions.
I think it'll end up looking fine with multiple buttons, each having a different color will help prevent misclicks (when compared to standalone icons).
If you want to jazz them up, I have two suggestion:
1. we have outline style buttons which could reduce the repeating color block issue (i.e. `btn-outline-success` class etc. for -info, -danger, -warning etc.)
2. you can add the icon in the button too